sdk/storage/azblob/blob/client.go

```diff
@@ -231,6 +231,30 @@ func (b *Client) GetTags(ctx context.Context, options *GetTagsOptions) (GetTagsR
 
 }
 
+// SetImmutabilityPolicy operation enables users to set the immutability policy on a blob.
+// https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-storage-overview
+func (b *Client) SetImmutabilityPolicy(ctx context.Context, options *SetImmutabilityPolicyOptions) (SetImmutabilityPolicyResponse, error) {
```

Comment:
Having all the params in `options` implies that they're all optional. Is this really the case? What happens if a customer calls `SetImmutabilityPolicy(ctx, nil)`? If this is legal, then the default behavior needs to be documented.

Reply:
So, the mode defaults to unlocked (can be nil). However, we do need to provide an expiry time. So, I changed that one parameter in the struct from optional to required.